lib/IRGen/GenDecl.cpp

```diff
@@ -6067,7 +6067,7 @@ IRGenModule::getResilienceExpansionForLayout(NominalTypeDecl *decl) {
 // variable's layout.
 ResilienceExpansion
 IRGenModule::getResilienceExpansionForLayout(SILGlobalVariable *global) {
-  if (hasPublicVisibility(global->getLinkage()))
+  if (hasPublicVisibility(global->getLinkage(), global->getModule().getOptions().EnableSerializePackage))
```

Comment:
This is wrong: if package-cmo is enabled then package things (types, variables, functions) have ResilienceExpansion::Maximal, because within a package they don't need to be resilient.